src/Components/Components/src/NavigationManager.cs

```diff
@@ -203,7 +205,15 @@ public virtual void Refresh(bool forceReload = false)
 
     private void NotFoundCore()
     {
-        _notFound?.Invoke(this, new NotFoundEventArgs());
+        if (_notFound == null)
+        {
+            // global router doesn't exist, no events were registered
+            throw new InvalidOperationException("No handler is subscribed to the OnNotFound event. Ensure that the application has a global router or manually subscribe to the OnNotFound event.");
+        }
```

Comment:
nit: I don't think this is blocking, but we shouldn't throw for this case, noop-ing is fine.